File: app/src/main/java/com/babic/filip/movieshack/database/DatabaseImpl.java
package com.babic.filip.movieshack.database;

import com.babic.filip.movieshack.database.dao.MovieDao;
import com.babic.filip.movieshack.model.Movie;

import java.util.List;

import javax.inject.Inject;

public class DatabaseImpl implements DatabaseInterface {

    private final MovieDao movieDao;

    @Inject
    public DatabaseImpl(final MovieDao movieDao) {
        this.movieDao = movieDao;
    }

    @Override
    public List<Movie> getMoviesByType(final String type) {
        return movieDao.getMoviesByType(type);
    }

    @Override
    public void addMovies(final List<Movie> movies) {
        for (final Movie movie : movies) {
            movieDao.addMovie(movie);
        }
    }

    @Override
    public void updateMovie(final Movie movie) {
        movieDao.updateMovie(movie);
    }

    @Override
    public Movie getMovieById(final String id) {
        return movieDao.getMovieById(id);
    }

    @Override
    public void clearMoviesByType(final String type) {
        movieDao.deleteMoviesByType(type);
    }
}

File: app/src/main/java/com/babic/filip/movieshack/database/DatabaseImpl.kt
package com.babic.filip.movieshack.database

import com.babic.filip.movieshack.database.dao.MovieDao
import com.babic.filip.movieshack.model.Movie

class DatabaseImpl(private val movieDao: MovieDao) : DatabaseInterface {

    override fun getMoviesByType(type: String): List<Movie> = movieDao.getMoviesByType(type)

    override fun addMovies(movies: List<Movie>) = movies.forEach { movieDao.addMovie(it) }
    override fun updateMovie(movie: Movie) = movieDao.updateMovie(movie)

    override fun getMovieById(id: String): Movie = movieDao.getMovieById(id)

    override fun clearMoviesByType(type: String) = movieDao.deleteMoviesByType(type)
}
